spring-session/src/main/java/org/springframework/session/data/cassandra/CassandraSessionRepository.java

```diff
@@ -0,0 +1,419 @@
+import org.springframework.session.jdbc.JdbcOperationsSessionRepository;
+import org.springframework.util.Assert;
+
+
+/**
+ * A {@link org.springframework.session.SessionRepository} implementation that uses
+ * Spring's {@link CassandraOperations} to store sessions in cassandra. This
+ * implementation does not support publishing of session events.
+ * It does make use of Cassandra's
+ * <a href="https://docs.datastax.com/en/cql/3.1/cql/cql_using/use_expire_c.html">TTL feature</a>,
+ * automatically removing expired sessions.
+ * Cassandra's
+ * <a href="https://docs.datastax.com/en/cql/3.1/cql/cql_reference/batch_r.html">batch feature</a>
+ * is also used to ensure that sessions and their lookup entries are added atomically.
+ * <p>
+ * An example of how to create a new instance can be seen below:
+ *
+ * <pre class="code">
+ * CassandraOperations cassandraOperations = new CassandraTemplate(sesion);
+ *
+ * // ... configure cassandraOperations ...
+ *
+ *
+ * CassandraSessionRepository sessionRepository =
+ *         new CassandraSessionRepository(cassandraOperations);
+ * </pre>
+ *
+ * For additional information on how to create and configure
+ * {@link org.springframework.data.cassandra.core.CassandraTemplate}, refer to the
+ * <a href="http://docs.spring.io/spring/docs/current/spring-framework-reference/html/spring-data-tier.html">
+ * Spring Framework Reference Documentation</a>.
+ * <p>
+ * By default, this implementation uses <code>spring_session</code> and
+ * <code>spring_session_by_name</code> tables to store sessions. Note that the table
+ * name can be customized using the {@link #setTableName(String)} method. In that case the
+ * table used to store attributes will be named using the provided table name, suffixed
+ * with <code>_by_name</code>.
+ *
+ * The table definition can be described as below:
+ *
+ * <pre class="code">
+ * CREATE TABLE session (
+ *   id uuid PRIMARY KEY,
+ *   attributes map&lt;text, text&gt;,
+ *   creation_time bigint,
+ *   last_accessed bigint,
+ *   max_inactive_interval_in_seconds int);
+ *
+ * CREATE TABLE session_by_name (
+ *   principal_name text,
+ *   id uuid,
+ *   PRIMARY KEY (principal_name, id));
+ *
+ * </pre>
+ *
+ * @author Andrew Fitzgerald
+ */
+public class CassandraSessionRepository implements FindByIndexNameSessionRepository<CassandraSessionRepository.CassandraHttpSession> {
+
+	private static final Logger log = LoggerFactory.getLogger(CassandraSessionRepository.class);
+	private static final PrincipalNameResolver PRINCIPAL_NAME_RESOLVER = new PrincipalNameResolver();
+
+
+	/**
+	 * Default name of the cassandra table used to store sessions.
+	 */
+	public static final String DEFAULT_TABLE_NAME = "spring_session";
+
+	//Temporary until #557 is resolved, see commends on PrincipalNameResolver
+	private static final String SPRING_SECURITY_CONTEXT = "SPRING_SECURITY_CONTEXT";
+
+
+	private final CassandraOperations cassandraOperations;
```

Comment:
Injecting `CqlOperations` is sufficient for this kind of access. Future versions of Spring Data Cassandra will break up the inheritance `CassandraOperations` `->` `CqlOperations` so using `CqlOperations` is the more future proof approach.